node-placeholder-scaler/scaler/scaler.py

```diff
@@ -94,9 +285,52 @@ def main():
 
             # Generate deployment config based on our config
             for pool_name, pool_config in config["nodePools"].items():
-                replica_count = replica_count_overrides.get(
-                    pool_name, pool_config["replicas"]
+                pool_usable_resources = usable_resources_result.get(
+                    pool_config["nodeSelector"]["hub.jupyter.org/pool-name"], {}
+                )
+                logging.info(f"Processing the node pool: {pool_name} ... ")
+                node_placeholder_deployment_reduction = 0
+                for node, resources in pool_usable_resources.items():
+                    logging.info(f"Checking node {node} in pool {pool_name} ...")
+                    logging.info(
+                        f"Node {node} has {resources['cpu_free_ratio']:.2f} CPU free ratio and {resources['mem_free_ratio']:.2f} Memory free ratio."
+                    )
+                    # Check if a placeholder pod is running on this node
+                    if not placeholder_pod_running_on_node(
+                        node, namespace, label_selector
+                    ):
+                        cpu_free_ratio = resources["cpu_free_ratio"]
+                        mem_free_ratio = resources["mem_free_ratio"]
+                        if cpu_free_ratio > 0.2 and mem_free_ratio > 0.2:
```

Comment:
just a quick comment on this line:  it'd be nice if this was documented a bit better, but we can take care of this as we continue to open-source the project!

:)